src/test/java/dev/openfga/sdk/api/client/OpenFgaClientTest.java

```diff
@@ -180,6 +181,62 @@ public void createStore_withClientCredentials() throws Exception {
         assertEquals(DEFAULT_STORE_NAME, response2.getName());
     }
 
+    @Test
+    public void createStore_withClientCredentialsWithRefresh() throws Exception {
+        // Given
+        String apiTokenIssuer = "oauth2.server";
+        String clientId = "some-client-id";
+        String clientSecret = "some-client-secret";
+        String apiToken = "some-generated-token";
+        String apiAudience = "some-audience";
+        clientConfiguration.credentials(new Credentials(new ClientCredentials()
+                .clientId(clientId)
+                .clientSecret(clientSecret)
+                .apiTokenIssuer(apiTokenIssuer)
+                .apiAudience(apiAudience)));
+        fga.setConfiguration(clientConfiguration);
+
+        String expectedBody = String.format("{\"name\":\"%s\"}", DEFAULT_STORE_NAME);
+        String requestBody = String.format("{\"id\":\"%s\",\"name\":\"%s\"}", DEFAULT_STORE_ID, DEFAULT_STORE_NAME);
+        mockHttpClient
+                .onPost(String.format("https://%s/oauth/token", apiTokenIssuer))
+                .withBody(allOf(
+                        containsString(String.format("client_id=%s", clientId)),
+                        containsString(String.format("client_secret=%s", clientSecret)),
+                        containsString(String.format("audience=%s", apiAudience)),
+                        containsString(String.format("grant_type=%s", "client_credentials"))))
+                .doReturn(200, String.format("{\"access_token\":\"%s\",\"expires_in\":\"%s\"}", apiToken, 1));
+        mockHttpClient
+                .onPost("https://localhost/stores")
```

Comment:
```suggestion
                .onPost("https://localhost/stores")
```

instead of using `https://localhost`, can you please use `https://api.fga.example`? We use it everywhere in the SDKs and it's a know URL that will never resolve if we accidentally forgot to mock

Reply:
Localhost is used everywhere in the testing for this SDK, can we potentially split that out to a separate PR?

Reply:
Split into https://github.com/openfga/java-sdk/pull/77 due to the larger diff it creates